cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractDistributedProgramRunner.java

```diff
@@ -366,6 +366,10 @@ private File saveCConf(CConfiguration conf, File file) throws IOException {
     // by the distributed ProgramRunner.
     CConfiguration copied = CConfiguration.copy(conf);
     copied.unset(Constants.AppFabric.RUNTIME_EXT_DIR);
+
+    // Set the CFG_LOCAL_DATA_DIR to a relative path as the data directory for the container should be relative to the
+    // container directory
+    copied.set(Constants.CFG_LOCAL_DATA_DIR, "data");
```

Comment:
Should we also update the description of the following property in the dap-default.xml to be more explicit that in distributed mode this configuration will not have any affect?
```java
  <property>
    <name>local.data.dir</name>
    <value>data</value>
    <description>
      Data directory for standalone mode
    </description>
  </property>
```